packages/client/wallets/smart-wallet/src/blockchain/wallets/account/passkey.ts

```diff
@@ -137,18 +137,35 @@ export class PasskeyCreationStrategy implements AccountCreationStrategy {
         return error;
     }
 
-    private decorate<Account extends KernelSmartAccount<EntryPoint>>(account: Account, passkeyName: string): Account {
+    private decorate<Account extends KernelSmartAccount<EntryPoint>>(
+        account: Account,
+        passkeyName: string,
+        walletParams: WalletParams
+    ): Account {
         return new Proxy(account, {
             get: (target, prop, receiver) => {
                 const original = Reflect.get(target, prop, receiver);
                 if (typeof original !== "function" || typeof prop !== "string" || !isAccountSigningMethod(prop)) {
                     return original;
                 }
 
+                const signer = walletParams.signer as PasskeySigner;
                 return async (...args: any[]) => {
+                    const isFirstTransaction = args?.[0]?.factoryData != null;
```

Comment:
I'm not aware how this will be called so I'm not sure why this means it is a first transaction?

Reply:
It's just one way to determine if the wallet has every minted anything, kinda strange but works